Project 4/SPARQL-JV.md

```diff
@@ -0,0 +1,389 @@
+
+SELECT ?tallShelf ?salePriceValueUSD
+WHERE {
+  ?shelf ro:instance_of homedev:Shelf ;
+           ro:bearer_of ?function ;
+           ro:composed_primarily_ of ?material ;
+           homedev:has_sale_price ?salePriceValueUSD .
+  FILTER (?function != homedev:twoShelf && 
+          ?salePriceValueUSD < 150 && 
+          ?material = homedev:woodMaterial && 
+          ?material != homedev:metalwireMaterial)
+}
+```
+**Problem 5-2.**
+- The federal department of education gives special grants to those educational institutions that garner the highest enrollments in the state (only one per state), whether that school is public or private. The total enrollment includes both undergraduate and postgraduate students. Find the college or university that would be eligible for such a grant in the state of California.
+- Give the name of the institution and number of students of a post-secondary educational institution located in California with the highest enrollment of students (undergrad and grad together).
+```
+PREFIX onto-ed: <https://www.nj.gov/admin/oit/ontology/>
+
+SELECT ?institution (MAX(?totalEnrollment) AS ?topEnrollment)
+WHERE {
+  ?institution a onto-ed:EducationalInstitution ;
+               onto-ed:type "PostsecondaryEducation" ;
+               onto-ed:located_in_state "California" ;
+               onto-ed:enrollment_number ?postgradEnrollment ;
+               onto-ed:enrollment_number ?undergradEnrollment ;
+               BIND (?postgradEnrollment + ?undergradEnrollment AS ?totalEnrollment) .
+}
+```
+**Problem 5-3.**
+Register a block of unused VINs.
+
+Take fields from multiple tables to generate a unique VIN. The third field is a check-digit, which in this case will be "9". Your manufacturing plant will only be assigning 1000. Lastly, verify that the output is exactly 17 characters long.
+
+The order of VIN construction: 
+
+WMI (ontomobile:worldManufacturerIdentifier)
+Vehicle attributes (ontomobile
+Check digit
+Model year
+Plant code
+Sequential number
+
+
+**Problem 5-4.**
+commerce: lettuce that was transported through AR or NE at risk for E Coli.
+construct: "recall" status for all such lettuce.
+- borked: to be filled in
+```
+borked
+```
+
+
+**Problem 5-5.**
+medical: person has genotype (string)
+         person has genotype (string)
+    construct: person has_predisposition testicularcancer
+- borked: to be filled in
+```
+borked
+``` 
+    
+
+**Problem 4-1.**
+- Around the world there are large buildings that bear the same shape as the famous Egyptian pyramids. These are called ziggurats. However, Egyptians were not the first to build this kind of structure. Which structures have the greatest estimated age?
+- Display the name of the oldest 25 ziggurats, current country location, name of associated culture when it was formed, and estimated age. Order by estimated age, descending (or date of completion ascending).
+```
+PREFIX anthro: <http://anthropologywiki.org/schema/ontology/>
+
```

Comment:
I wouldn't say this is a level 4, it is an interesting query but for level 4s I am thinking about things like using many different functions, creating graphs, doing math, etc. You do use order by asc and filter that are interesting but if I think of the level 4s in codewars this looks easier